test/index.test.js

```diff
@@ -780,8 +780,19 @@ describe('Segment.io', function() {
         segment.options.crossDomainIdServers = [
           'userdata.example1.com',
           'xid.domain2.com',
-          'localhost'
+          'xid.domain3.com'
         ];
+        segment._getCurrentHostname = function() {
+          return 'example1.com';
+        };
+        segment._getTld = function(domain) {
+          if (domain === 'example1.com' || domain === 'userdata.example1.com') {
+            return 'example1.com';
+          }
+          // Mimic topDomain which returns "" if you check a domain you're not
+          // currently on.
+          return '';
```

Comment:
Do we have a test for this case somewhere? I'm not seeing it but maybe I'm not looking hard enough